Source/WebCore/page/Navigation.cpp

```diff
@@ -739,14 +743,29 @@ bool Navigation::innerDispatchNavigateEvent(NavigationNavigationType navigationT
             if (!failure) {
                 dispatchEvent(Event::create(eventNames().navigatesuccessEvent, { }));
 
-                // FIXME: 7. If navigation's transition is not null, then resolve navigation's transition's finished promise with undefined.
-                m_transition = nullptr;
+                if (m_transition) {
+                    auto promiseValue = m_transition->finished().promise();
+                    auto& jsPromise = *JSC::jsCast<JSC::JSPromise*>(promiseValue);
+                    auto& globalObject = *JSC::jsCast<JSDOMGlobalObject*>(jsPromise.globalObject());
+                    auto deferredFinished = DeferredPromise::create(globalObject, jsPromise);
+                    deferredFinished->resolve();
```

Comment:
This change doesn't make sense to me.  We are making a DeferredPromise, never attaching it to anything, then resolving and destroying it immediately.  That seems like it's not hooked up right.  Same below.

Reply:
I had a different take on it, if you think this is cleaner I can open a PR: https://github.com/TingPing/WebKit/commit/20400d93592520ae8110d0c324f80deed7076dc6

Reply:
Yes, that looks good, please go ahead. Note that I think it lacks the error case where the promise is rejected. Also, could it work using DOMPromiseProxy (not sure)?